test/addons-napi/test_reference/test.js

```diff
@@ -11,80 +11,110 @@ const test_reference = require(`./build/${common.buildType}/test_reference`);
 // of a finalizer callback increments the finalizeCount property.
 assert.strictEqual(test_reference.finalizeCount, 0);
 
-{
-  // External value without a finalizer
-  let value = test_reference.createExternal();
-  assert.strictEqual(test_reference.finalizeCount, 0);
-  assert.strictEqual(typeof value, 'object');
-  test_reference.checkExternal(value);
-  value = null;
-  global.gc();
-  assert.strictEqual(test_reference.finalizeCount, 0);
+// Run each test function in sequence,
+// with an async delay and GC call between each.
+function runTests(i, title, tests) {
+  if (tests[i]) {
+    if (typeof tests[i] === 'string') {
+      title = tests[i];
+      runTests(i + 1, title, tests);
+    } else {
+      try {
+        tests[i]();
+      } catch (e) {
+        console.error('Test failed: ' + title);
+        throw e;
+      }
+      setImmediate(() => {
+        global.gc();
+        runTests(i + 1, title, tests);
+      });
+    }
+  }
 }
+runTests(0, undefined, [
 
-{
-  // External value with a finalizer
-  let value = test_reference.createExternalWithFinalize();
-  assert.strictEqual(test_reference.finalizeCount, 0);
-  assert.strictEqual(typeof value, 'object');
-  test_reference.checkExternal(value);
-  value = null;
-  global.gc();
-  assert.strictEqual(test_reference.finalizeCount, 1);
-}
-
-{
-  // Strong reference
-  let value = test_reference.createExternalWithFinalize();
-  assert.strictEqual(test_reference.finalizeCount, 0);
-  test_reference.createReference(value, 1);
-  assert.strictEqual(test_reference.referenceValue, value);
-  value = null;
-  global.gc(); // Value should NOT be GC'd because there is a strong ref
-  assert.strictEqual(test_reference.finalizeCount, 0);
-  test_reference.deleteReference();
-  global.gc(); // Value should be GC'd because the strong ref was deleted
-  assert.strictEqual(test_reference.finalizeCount, 1);
-}
-
-{
-  // Strong reference, increment then decrement to weak reference
-  let value = test_reference.createExternalWithFinalize();
-  assert.strictEqual(test_reference.finalizeCount, 0);
-  test_reference.createReference(value, 1);
-  value = null;
-  global.gc(); // Value should NOT be GC'd because there is a strong ref
-  assert.strictEqual(test_reference.finalizeCount, 0);
+  'External value without a finalizer',
```

Comment:
I was thinking it would actually be easier/less code, but if you don't think that's the case I don't want to hold up getting this landed.  I may take a look/PR later on when I have time if what I was thinking can simply, if not then I'll leave it be.